### DISCUSSION TITLE
chinese whisper cluster edges example

### DISCUSSION THREAD

**Discussion by Compaile** [davisking/dlib#2776] [General]
Hi,
im not 200% sure if its a bug or something i dont quite understand.

In the examples for the chinese cluster algorithm
(https://github.com/davisking/dlib/blob/master/examples/dnn_face_recognition_ex.cpp)

why do all examples build edges for edge node to themself 

` for (size_t i = 0; i < face_descriptors.size(); ++i)
    {
        for (size_t j = i; j < face_descriptors.size(); ++j)
        {
            // Faces are connected in the graph if they are close enough.  Here we check if
            // the distance between two face descriptors is less than 0.6, which is the
            // decision threshold the network was trained to use.  Although you can
            // certainly use any other threshold you find useful.
            if (length(face_descriptors[i]-face_descriptors[j]) < 0.6)
                edges.push_back(sample_pair(i,j));
        }
    }`

in other words why isnt it  
`for (size_t j = i+1; j < face_descriptors.size(); ++j)`

Thanks!

**Comment by davisking** [comment]
It doesn't matter.  You can do it either way.

**Reply by Compaile** [reply]
alright thanks !
